### PR TITLE
Change parameter of SetCurrentThreadName to string_view

### PR DESCRIPTION
**Pull request by dimitry-**
Also slightly optimize Linux implementation

Test: run OrbitBaseTests

### CODE REVIEW

File: src/OrbitBase/ThreadUtilsLinux.cpp [hunk 1]
```diff
@@ -40,12 +40,15 @@ std::string GetThreadName(pid_t tid) {
   return comm_content.value();
 }
 
-void SetCurrentThreadName(const std::string& thread_name) {
-  std::string name = thread_name;
-  if (name.length() >= kMaxThreadNameLength) {
-    name[kMaxThreadNameLength - 1] = '\0';
+void SetCurrentThreadName(std::string_view thread_name) {
+  std::string buf;
+  const char* thread_name_cstr = thread_name.data();
```
Comment (pierricgimmig) [inline src/OrbitBase/ThreadUtilsLinux.cpp:45]: thread_name_cstr won't be null terminated which is expected by pthread_setname_np expects

File: src/OrbitBase/ThreadUtilsLinux.cpp [hunk 2]
```diff
@@ -40,12 +40,15 @@ std::string GetThreadName(pid_t tid) {
...
+  if (thread_name.length() >= kMaxThreadNameLength) {
+    buf = thread_name.substr(0, kMaxThreadNameLength - 1);
+    thread_name_cstr = buf.data();
```
Comment (pierricgimmig) [inline src/OrbitBase/ThreadUtilsLinux.cpp:48]: same thing here.

**Review by pierricgimmig — COMMENTED** [review]
Hi Dimitry, thanks for this, I think there are issues with the strings not being null terminated however.

File: src/OrbitBase/ThreadUtilsLinux.cpp (hunk 2 above)
Comment (dimitry-) [reply src/OrbitBase/ThreadUtilsLinux.cpp:48]: well, this part is fine because std::strings c_str() are null-terminated.

Comment (pierricgimmig) [reply src/OrbitBase/ThreadUtilsLinux.cpp:48]: well c_str() yes, but not data() I think, no?

File: src/OrbitBase/ThreadUtilsLinux.cpp (hunk 1 above)
Comment (dimitry-) [reply src/OrbitBase/ThreadUtilsLinux.cpp:45]: right. fixed.

File: src/OrbitBase/ThreadUtilsLinux.cpp (hunk 2 above)
Comment (pierricgimmig) [reply src/OrbitBase/ThreadUtilsLinux.cpp]: actually it seems data does return a null terminated string https://en.cppreference.com/w/cpp/string/basic_string/data#:~:text=The%20returned%20array%20is%20null,to%20a%20single%20null%20character.&text=The%20pointer%20obtained%20from%20data,any%20standard%20library%20function%2C%20or

**Review by pierricgimmig — APPROVED** [review]
LGTM!

**Pull request merged**